gcloud/datastore/test_query.py

```diff
@@ -543,6 +542,50 @@ def test___iter___w_more(self):
         self.assertEqual(connection._called_with[0], EXPECTED1)
         self.assertEqual(connection._called_with[1], EXPECTED2)
 
+    def test___iter___w_limit(self):
+        from gcloud.datastore.query import _pb_from_query
+
+        connection = _Connection()
+        client = self._makeClient(connection)
+        query = _Query(client, self._KIND, self._PROJECT, self._NAMESPACE)
+        skipped_results = 4
+        self._addQueryResults(connection, more=True,
+                              skipped_results=skipped_results)
+        self._addQueryResults(connection)
+        offset = 13
+        iterator = self._makeOne(query, client, limit=2, offset=offset)
+        entities = list(iterator)
+
+        self.assertFalse(iterator._more_results)
+        self.assertEqual(len(entities), 2)
+        for entity in entities:
+            self.assertEqual(
+                entity.key.path,
+                [{'kind': self._KIND, 'id': self._ID}])
+            self.assertEqual(entities[1]['foo'], u'Foo')
+        qpb1 = _pb_from_query(query)
+        qpb1.limit.value = 2
+        qpb1.offset = offset
+        qpb2 = _pb_from_query(query)
+        qpb2.start_cursor = self._END
+        qpb2.limit.value = 1
```

Comment:
Well either that or if only 4 results are skipped in `qpb1` then it shouldn't return any results (and `qpb2` should still have limit=2). It may be worth it to just add in a third query to test everything:

User query: LIMIT 2 OFFSET 13

1: skipped: 4
2: skipped: 9 results: 1
3: results: 1